File: articles/the-unknown-the-variable-and-islamic-mathematical-philosophy/cpp/square_completion.cpp
#include <iostream>

int main() {
    double b = 10.0;
    double c = 39.0;
    double completion = (b / 2.0) * (b / 2.0);
    std::cout << "completion_term=" << completion << "\n";
    std::cout << "completed_rhs=" << c + completion << "\n";
    return 0;
}

File: articles/the-unknown-the-variable-and-islamic-mathematical-philosophy/rust/square_completion.rs
fn main() {
    let b = 10.0_f64;
    let c = 39.0_f64;
    let completion = (b / 2.0).powi(2);
    println!("completion_term={:.6}", completion);
    println!("completed_rhs={:.6}", c + completion);
}
